app/src/main/java/android/coding/interview/makeitawesome/adapter/PhotosAdapter.java

```diff
@@ -35,11 +64,34 @@ public int getItemCount() {
         return mData.size();
     }
 
-    public static class ViewHolder extends RecyclerView.ViewHolder {
-        final TextView mText;
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private final TextView mTitle;
+        private final ImageView mImageView;
+
         public ViewHolder(View itemView) {
             super(itemView);
-            mText = (TextView) itemView;
+            mTitle = (TextView) itemView.findViewById(R.id.tv_image_title);
+            mImageView = (ImageView) itemView.findViewById(R.id.iv_image);
+            itemView.setOnClickListener(this);
+        }
+
+        public TextView getImageTitle() {
+            return mTitle;
+        }
+
+        public ImageView getIv_image() {
+            return mImageView;
         }
+
+        @Override
+        public void onClick(View v) {
+            if (clickListener!= null){
+                clickListener.itemClicked(v, getAdapterPosition());
+            }
```

Comment:
Oh, no, no. I misunderstood what he was doing. :(

I thought he created a new interface to notify "events", but he is accesing a clickListener in the Adapter, totally coupling both classes. It's not so nice.